sdk/trace/batch_span_processor_test.go

```diff
@@ -474,6 +495,46 @@ func TestBatchSpanProcessorForceFlushSucceeds(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestBatchSpanProcessorLogsWarningOnNewDropSpans(t *testing.T) {
+	tLog := testr.NewWithOptions(t, testr.Options{Verbosity: 1})
+	l := &logCounter{LogSink: tLog.GetSink()}
+	otel.SetLogger(logr.New(l))
+
+	te := testBatchExporter{}
+	tp := basicTracerProvider(t)
+	option := testOption{
+		name: "default BatchSpanProcessorOptions",
+		o: []sdktrace.BatchSpanProcessorOption{
+			sdktrace.WithMaxQueueSize(0),
+			sdktrace.WithMaxExportBatchSize(10),
+		},
+		genNumSpans: 100,
+	}
+	ssp := createAndRegisterBatchSPNonBlocking(option, &te)
```

Comment:
It doesn't seem like the `createAndRegisterBatchSPNonBlocking` function is needed. It just wraps a call to `NewBatchSpanProcessor`

```suggestion
	ssp := sdktrace.NewBatchSpanProcessor(&te, option.o...)
```